### ISSUE TITLE
Content Import and REST API- Allow remote urls to be specified for binary fields

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#16852]
One limitation of our content importer is that it cannot import data into binary fields unless the file is already uploaded into dotCMS as a file asset.  We need to allow users to specify a full URL as the value of a binary field which could then be retrieved by dotCMS and imported into the appropriate field in the content object.

1.  If the value of a binary field starts with `http:// || https://` we should use the TempResourceAPI.byUrl to download this file as a temp resource.  If the remote file is successfully downloaded, then the Content Importer should pass the newly created `temp_id`
as the value for that binary.  

2.  We will need a new TempResourceAPI method called `TempResourceAPI.testURL` which takes a url (starts with `http:// || https://`) and returns a `boolean` This method does an HTTP HEAD request against the passed in URL (following any redirects).  If the url results in a 200 response, then the URL is valid.  This method should be called by the import process when validating the import data, which should also report on the results of the test.
The `TempResourceAPI.byUrl file creation download process should only happen at the final stage of the Content Import process, when the content is being imported.

**Comment by fmontes** [comment]
## Questions
1. What happen if there are like 1000 images to download?

**Comment by wezell** [comment]
This is already done, no?

**Comment by wezell** [comment]
Sorry, I see, only a portion of this is done - if there are 1000 files to download, then we download the 1000 files and have 1000 temp resources ready to be imported.

**Comment by erickgonzalez** [comment]
Both methods are already created (validate the URL and download the file) and their respective tests, what is missing is add those methods to the content importer, so it is a binary field and the value is a full URL, download the file and associate it to the content.

**Comment by fmontes** [comment]
### Behavior:

1. We wait all downloads to happen before import (and create the temp resources)
2. We show errors in the requests and show the user

### Questions
1. What happen if the field variable have `allowedFileTypes` set and the CSV have a different type?
2. What happen if the field variable have `maxFileLength` set and the CSV have a bigger file size?

**Comment by fmontes** [comment]
One idea could be after the import in the results page add information about the fields that failed to import because `allowedFileTypes` and/or `maxFileLength`, like here:

![Screen Shot 2019-12-03 at 11 59 56](https://user-images.githubusercontent.com/751424/70076538-9a04ca80-15c4-11ea-8708-4febf958fa9c.png)

**Comment by fmontes** [comment]
Let's kicks this into a new card: https://github.com/dotCMS/core/issues/16852#issuecomment-561285450

**Comment by erickgonzalez** [comment]
Steps to test:
1- Create a CT with a Text and a Binary Field (title and file)
2- Create a CSV file like this:
```
title,file
t400,https://raw.githubusercontent.com/url/throws/400.jpg
t404,https://raw.githubusercontent.com/dotCMS/core/throws/dotCMS/404.jpg
tfalse,test://raw.githubusercontent.com/dotCMS/core/master/dotCMS/src/main/webapp/html/images/skin/logo.gif
t200,https://raw.githubusercontent.com/dotCMS/core/master/dotCMS/src/main/webapp/html/images/skin/logo.gif
```
3- Import the CSV

The 3 first lines should return with errors since those are not valid URL or do not return 200, the fourth line is fine, so only one content should be created. After the import is done, you should be able to see the created content and the file.

**Comment by erickgonzalez** [comment]
PR: https://github.com/dotCMS/core/pull/17703

**Comment by bryanboza** [comment]
When you try to import an invalid piece of content we have too many lines of error in the log file, we need to catch this exception and print something clear about the problem, also maybe this should be a WARN
Current ST: https://gist.github.com/bryanboza/7592c753b0805aaa1479bb432cd05c1a

**Comment by erickgonzalez** [comment]
PR: https://github.com/dotCMS/core/pull/17910
Log has been reduced and changed to WARN, we were logging the error twice now it's logging only once. This can't be reduced anymore since the ST is being logged by the **BadRequestException** when it's being created 
https://github.com/dotCMS/core/blob/ae1186f3766588cf27555cfe660d9768cba2b529/dotCMS/src/main/java/com/dotcms/http/CircuitBreakerUrl.java#L175
https://github.com/dotCMS/core/blob/68a3c9169f21015f18cfc43f7bf1acefcf87946c/dotCMS/src/main/java/com/dotcms/rest/exception/HttpStatusCodeException.java#L43
I tried to catch this exception but seems that is being thrown by the failsafe jar.

**Comment by fmontes** [comment]
Logs reduced for 5.2.5

**Comment by bryanboza** [comment]
Fixed, now we have a WARN instead errors... Tested on release-5.2.5 // Postgres // FF
